File: FrontEnd/aurofin-frontend/src/app/entity/ApplicationResponse.ts
import { LoanResponseDto } from './LoanResponseDto';
import { InstallmentDto } from './InstallmentDto';

export interface ApplicationResponse {
  status: string;
  rejectionReason?: string;
  requestedAmount: number;
  requestedTenure: number;
  totalLoanAmount: number;
  createdAt: string; // ISO date string
  loanResponseDto: LoanResponseDto;
  installmentDtos: InstallmentDto[];
  managerName: string;
  applicantAge: number;
}

File: Backend/src/main/java/com/lendingApp/main/dto/ApplicationResponse.java
package com.lendingApp.main.dto;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ApplicationResponse {
	private String status; // pending / approved / rejected
	private String rejectionReason;
    private Double requestedAmount;
    private Integer requestedTenure;
    private Double totalLoanAmount;
    private LocalDateTime createdAt;
    private LoanResponseDto loanResponseDto;
    private List<InstallmentDto> installmentDtos;
    private String managerName;
    private Integer applicantAge;
}
